inc/pantheon-updates.php

```diff
@@ -1,119 +1,186 @@
-				<br />
-				<?php echo $update_help; ?>
-			</div>
-			<?php
+		if ( 'update-core' === $screen->id || 'update-core-network' === $screen->id ) {
+			// Escaping is handled above when we're buffering the output, so we can ignore it here.
+			echo $notice_html; // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
 		}
 	} else {
-		// If WP core is out of date, alter the message and show the nag everywhere.
-		// Translators: %s is a URL to the user's Pantheon Dashboard.
-		$notice_message = wp_kses_post( sprintf( __( 'A new WordPress update is available! Please update from <a href="%s">your Pantheon dashboard</a>.', 'pantheon-systems' ), 'https://dashboard.pantheon.io/sites/' . $_ENV['PANTHEON_SITE'] ) );; ?>
-		<div class="<?php echo $div_class; ?>" style="<?php echo $div_style; ?>">
-			<p style="<?php echo $paragraph_style; ?>">
-				<?php echo $notice_message; ?>
-			</p>
-			<?php echo $upstream_help_message; ?>
-			<br />
-			<?php echo $update_help; ?>
-		</div>
-		<?php
+		echo $notice_html; // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
 	}
 }
 
-// Register Pantheon specific WordPress update admin notice
-add_action( 'admin_init', '_pantheon_register_upstream_update_notice' );
-function _pantheon_register_upstream_update_notice(){
-	// but only if we are on Pantheon
-	// and this is not a WordPress Ajax request
-	if( isset( $_ENV['PANTHEON_ENVIRONMENT'] ) && ! wp_doing_ajax() ){
+/**
+ * Register Pantheon specific WordPress update admin notice.
+ *
+ * @return void
+ */
+function _pantheon_register_upstream_update_notice() {
+	// Only register notice if we are on Pantheon and this is not a WordPress Ajax request.
+	if ( isset( $_ENV['PANTHEON_ENVIRONMENT'] ) && ! wp_doing_ajax() ) {
 		add_action( 'admin_notices', '_pantheon_upstream_update_notice' );
 		add_action( 'network_admin_notices', '_pantheon_upstream_update_notice' );
 	}
 }
+add_action( 'admin_init', '_pantheon_register_upstream_update_notice' );
 
-// Return zero updates and current time as last checked time
-function _pantheon_disable_wp_updates() {
-	include ABSPATH . WPINC . '/version.php';
-	return (object) array(
-		'updates' => array(),
-		'version_checked' => $wp_version,
+/**
+ * Return zero updates and current time as last checked time.
+ *
+ * @return object
+ */
+function _pantheon_disable_wp_updates() : object {
+	$wp_version = _pantheon_get_current_wordpress_version();
+	return (object) [
+		'updates' => [],
+		'version_checked' => $wp_version, // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UndefinedVariable
 		'last_checked' => time(),
-	);
+	];
 }
 
 // In the Test and Live environments, clear plugin/theme update notifications.
 // Users must check a dev or multidev environment for updates.
-if ( in_array( $_ENV['PANTHEON_ENVIRONMENT'], array('test', 'live') ) && (php_sapi_name() !== 'cli') ) {
+if ( in_array( $_ENV['PANTHEON_ENVIRONMENT'], [ 'test', 'live' ], true ) && ( php_sapi_name() !== 'cli' ) ) {
```

Comment:
git diff may be confusing me here, but does this need to check `isset( $_ENV['PANTHEON_ENVIRONMENT']` or is that handled with line 9 above?

Reply:
we didn't before, but that's a good callout